dartagnan/src/main/java/com/dat3m/dartagnan/program/event/common/FenceBase.java

```diff
@@ -2,20 +2,19 @@
 
 import com.dat3m.dartagnan.program.event.Tag;
 import com.dat3m.dartagnan.program.event.core.AbstractEvent;
+import com.google.common.base.Preconditions;
 
 @NoInterface
 public abstract class FenceBase extends AbstractEvent {
 
     protected final String name;
-    protected final String mo; // May be NULL or empty for fences that do not support a mo.
+    protected final String mo; // For fences that do not support a mo, use Fence instead.
 
     public FenceBase(String name, String mo) {
+        Preconditions.checkArgument(!mo.isEmpty());
         this.name = name;
         this.mo = mo;
-        this.addTags(Tag.VISIBLE, Tag.FENCE);
-        if (mo != null && !mo.isEmpty()) {
-            addTags(mo);
-        }
+        addTags(Tag.VISIBLE, Tag.FENCE, mo);
```

Comment:
Why do we require a memory order now? I think fences without a `mo` are reasonable.